unbricking.md

```diff
@@ -0,0 +1,193 @@
+CPU Clock: 400 MHz
+Net:   SGMII, SERDES [PRIME]
+Press SPACE to delay and Ctrl-C to abort autoboot in 5 seconds
+FALCON =>
+```
+Press Ctrl-C to abort the boot to get u-boot access.
+
+If the bootloader is still locked there is no option to interrupt the boot (or even this level of boot message). In that case see [Unbrick
+Unprepared SFP Modules](#unlock_bricked).
+
+## Fixing Environment Issues
+
+If the bricking was caused by a bad u-boot environment it can be fixed using
+``setenv``. After changing the environment use ``saveenv`` to make the changes
+permanent.
+
+For example, if the active image 1 is not bootable, but image 0 is still
+bootable then use:
+```
+setenv commit 0
+saveenv
+```
+
+## Replacing Firmware
+
+Credit to anon23891239 describing it [here](https://forum.openwrt.org/t/support-ma5671a-sfp-gpon/48042/24).
+
+If no bootable firmware image is left the images can be replaced via u-boot.
+The two images are stored at 0xC0000(length 0x740000) and 0x800000(length
+0x800000) respectively.
+
+To upload a firmware image onto the machine use a kermit transfer:
+```
+FALCON => loadb 0x80800000
+## Ready for binary (kermit) download to 0x80800000 at 115200 bps...
+```
+
+To replace the first image and activate it use:
+```
+FALCON => sf probe 0
+FALCON => sf erase C0000 740000
+FALCON => sf write 80800000 C0000 740000
+FALCON => setenv commit 0
+```
+
+## Unbrick Unprepared SFP Modules
+<a name="unlock_bricked"></a>
+
+If a SFP module is bricked but it is not prepared with the above then it's still
+possible to unbrick, but it requires opening the module and temporarily shorting
+pins to cause the regular boot to fail.
+
+Credit to hakan describing this [here](https://forum.openwrt.org/t/will-gpon-nokia-g-010s-a-change-sn/69602).
+
+### Short the Flash and Boot
+
+Pins 4 and 5 on the flash chip need to he shorted temporarily to interrupt the
+boot process (e.g., with some tweezers). Then connect the STP to the UART board.
+
+With the pins shortened the SFP module will not be able to read the boot loader
+from the flash chip, making it fall back into a recovery mode that allows
+uploading a new boot loader.
+
+![Flash Image](images/flash_short.png)
+
+Once the boot prompt is visible remove the short between the pins.
+
+### Upload temp Temp Loader
+
+With the shorted flash chip the boot ends up in the recovery console:
+```
+ROM: V1.1.4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: CFG 0x00000006
+ROM: SFLASH-4
+ROM: Boot? (0-9A-F<CR>)
+```
+Enter 7 as the boot option, this will start an XMODEM transfer.
+```
+ROM: CFG 0x00000007
+ROM: XMODEM
+CCCC
+```
+
+Upload [1224ABORT.bin](https://ma5671a.s3.nl-ams.scw.cloud/1224ABORT.bin) via
```

Comment:
We should add the file to the repo, it's better to ensure the link doesn't become dead.

Reply:
done